### ISSUE TITLE
Work on function/arithmetic priorities

### ISSUE THREAD

**Issue by oleobal** [oleobal/rockstar-interpreter#10]

**Comment by zebralt** [comment]
I think we need to overhaul the `expression` tokenization process. We should put as many unknown elements in a row as we can in an `expression` node int the AST, to be able to parse them properly and apply operator priority, while keeping on subdivising expressions.

```
Aa plus Fibonacci taking Bb
[{'type' : 'expression', 'value' : 'Aa'}, -> evaluated to variable
 {'type' : 'arithmetic operator', 'value' : 'ADD'},
 {'type' : 'expression', 'value' : 'Fibonacci taking Bb'}] -> evaluated to function call
```
and so on. That's what we've been on about for a while now, but I think it's still not fully formalized in the code.

Also related to #9 .

**Comment by oleobal** [comment]
This is true, there's no explicit priority and we're going to have to think of a way to express it which will probably  involve refactoring the tokenizer.

(Maybe we should have used a lexer..)

I know it's not exactly a simple task. What do you think ? Should we plan on doing it as soon as possible (it is a sizeable issue), or push it back a bit until we have something that works for fizzbuzz and al so we can publish it ?

Edit : also, no way we can show that to Voisin with the current tokenizer. He'd probably be ashamed.
